API_GUIDE.md

```diff
@@ -45,23 +45,26 @@ for data, target in train_loader:
   loss = loss_fn(output, target)
   loss.backward()
   xm.optimizer_step(optimizer)
```

Comment:
It should be:
```Python
xm.optimizer_step(optimizer, barrier=True)
```

Reply:
I thought about that but then I also thought there may be value in exposing `mark_step()` in the guide. If I do `barrier=True` then we won;t mention mark_step at all. wdyt?

Reply:
No please. Let's not ask user to add more lines and understand more offline APIs for the common case.

Reply:
alternative is, include `barrier=True` here and delete that in the multicore cases, and make a note.